lib/src/components/NxTransferList/__tests__/TransferListHalf.test.tsx

```diff
@@ -273,7 +277,149 @@ describe('TransferListHalf', function() {
     });
   });
+
+  it('passes onReorderItem with id and direction when move up or down button is clicked', function() {
+    const items = [{
+      id: 1,
+      displayName: 'a'
+    }, {
+      id: 2,
+      displayName: 'b'
+    }, {
+      id: 3,
+      displayName: 'c'
+    }];
+
+    const onReorderItem = jest.fn(),
+        component = getMounted({ allowReordering: true, items, onReorderItem }),
+        list = component.find('.nx-transfer-list__item-list');
+
+    expect(onReorderItem).not.toHaveBeenCalled();
+
+    const firstItem = list.find('.nx-transfer-list__item').at(0),
+        secondItem = list.find('.nx-transfer-list__item').at(1);
+    const moveDownButton = firstItem.find('.nx-btn').at(1);
+    const moveUpButton = secondItem.find('.nx-btn').at(0);
+
+    expect(moveDownButton).toExist();
+    expect(moveUpButton).toExist();
+
+    moveDownButton.simulate('click');
+
+    expect(onReorderItem).toHaveBeenCalledWith(0, 1);
+
+    moveUpButton.simulate('click');
+
+    expect(onReorderItem).toHaveBeenCalledWith(1, -1);
+  });
+
+  it('set the move up and down icon button with faArrowUp and faArrowDown respectively',
+      function() {
+        const items = [{
+          id: 1,
+          displayName: 'a'
+        }, {
+          id: 2,
+          displayName: 'b'
+        }, {
+          id: 3,
+          displayName: 'c'
+        }];
+
+        const component = getMounted({ allowReordering: true, items }),
+            listItem = component.find('.nx-transfer-list__item').at(0);
+
+        const buttons = listItem.find('.nx-transfer-list__button-bar .nx-btn');
+        const moveUpButton = buttons.at(0).find(NxFontAwesomeIcon);
+        const moveDownButton = buttons.at(1).find(NxFontAwesomeIcon);
+
+        expect(moveUpButton).toHaveProp('icon', faArrowUp);
+        expect(moveDownButton).toHaveProp('icon', faArrowDown);
+      }
+  );
+
+  it('sets move up and down buttons to disabled state when items are filtered', function() {
+    const items = [{
+      id: 1,
+      displayName: 'a1'
+    }, {
+      id: 2,
+      displayName: 'b'
+    }, {
+      id: 3,
+      displayName: 'c1'
+    }];
+
+    const onReorderItem = jest.fn(),
+        component = getMounted({ filterValue: '1', allowReordering: true, items, onReorderItem }),
+        list = component.find('.nx-transfer-list__item-list');
+
+    expect(onReorderItem).not.toHaveBeenCalled();
+
+    const secondItem = list.find('.nx-transfer-list__item').at(1);
+    const moveUpButton = secondItem.find('.nx-btn').at(0);
+    const moveDownButton = secondItem.find('.nx-btn').at(1);
+
+    expect(moveUpButton).toExist();
+    expect(moveDownButton).toExist();
+
+    expect(moveUpButton.prop('disabled')).toBe(true);
+    expect(moveDownButton.prop('disabled')).toBe(true);
+
+    moveDownButton.simulate('click');
+
+    expect(onReorderItem).not.toHaveBeenCalled();
+
+    moveUpButton.simulate('click');
+
+    expect(onReorderItem).not.toHaveBeenCalled();
+  });
```

Comment:
Needs a test for the title logic

Reply:
> Needs a test for the title logic

Added test here: 5759f98d182e17a0e9d67472f4f8f68b2f8af0e5

Reply:
> Added test here: 5759f98

:+1: that'll need to be updated though as the approach has changed slightly

Reply:
> > Added test here: [5759f98](https://github.com/sonatype/sonatype-react-shared-components/commit/5759f98d182e17a0e9d67472f4f8f68b2f8af0e5)
> 
> 👍 that'll need to be updated though as the approach has changed slightly

Yup test updated: 885f3cb8f599364a04751640783fbcad7193f3df 😄